extensions/amp-access/0.1/amp-access.js

```diff
@@ -365,7 +373,40 @@ export class AccessService {
    * @private
    */
   handleAction_(invocation) {
-    log.fine(TAG, 'Invocation: ', invocation);
+    if (invocation.method == 'login') {
+      this.login();
+    }
+  }
+
+  /**
+   * Runs the Login flow. Returns a promise that is resolved if login succeeds
+   * or is rejected if login fails.
+   * @return {!Promise}
+   */
+  login() {
+    if (this.loginPromise_) {
+      return this.loginPromise_;
+    }
+    log.fine(TAG, 'Start login');
+    this.loginPromise_ = this.buildUrl_(this.config_.login).then(url => {
+      log.fine(TAG, 'Login URL: ', url);
+      return this.openLoginDialog_(url);
+    }).then(result => {
+      log.fine(TAG, 'Login dialog completed: ', result);
+      this.loginPromise_ = null;
+      const query = parseQueryString(result);
+      const s = query['success'];
+      const success = (s != 'false' && s != 'no' && s != '0');
+      if (success) {
+        // Repeat the authorization flow.
+        this.runAuthorization_();
+      }
+    }).catch(reason => {
+      log.fine(TAG, 'Login dialog failed: ', reason);
+      this.loginPromise_ = null;
+      throw reason;
```

Comment:
I was asking, because we should avoid throwing (and reporting to the error backend) on expected failures.

I believe window.open never fails, but it can return null. We should definitely add a null check.